v2/internal/signal/signal.go

```diff
@@ -1,73 +1,43 @@
 package signal
 
 import (
-	"context"
 	"os"
 	gosignal "os/signal"
 	"sync"
 	"syscall"
-
-	"github.com/wailsapp/wails/v2/internal/logger"
-	"github.com/wailsapp/wails/v2/internal/servicebus"
 )
 
-// Manager manages signals such as CTRL-C
-type Manager struct {
-	// Service Bus
-	bus *servicebus.ServiceBus
-
-	// logger
-	logger logger.CustomLogger
-
-	// signalChannel
-	signalchannel chan os.Signal
+var signalChannel = make(chan os.Signal, 2)
 
-	// ctx
-	ctx    context.Context
-	cancel context.CancelFunc
+var callbacks []func()
+var lock sync.Mutex
 
-	// Parent waitgroup
-	wg *sync.WaitGroup
-}
-
-// NewManager creates a new signal manager
-func NewManager(ctx context.Context, cancel context.CancelFunc, bus *servicebus.ServiceBus, logger *logger.Logger) (*Manager, error) {
-
-	result := &Manager{
-		bus:           bus,
-		logger:        logger.CustomLogger("Signal Manager"),
-		signalchannel: make(chan os.Signal, 2),
-		ctx:           ctx,
-		cancel:        cancel,
-		wg:            ctx.Value("waitgroup").(*sync.WaitGroup),
-	}
-
-	return result, nil
+func OnShutdown(callback func()) {
+	lock.Lock()
+	defer lock.Unlock()
+	callbacks = append(callbacks, callback)
 }
 
 // Start the Signal Manager
-func (m *Manager) Start() {
+func Start() {
 
 	// Hook into interrupts
-	gosignal.Notify(m.signalchannel, os.Interrupt, syscall.SIGTERM, syscall.SIGINT)
+	gosignal.Notify(signalChannel, os.Interrupt, syscall.SIGTERM, syscall.SIGINT)
 
-	m.wg.Add(1)
+	var wg sync.WaitGroup
```

Comment:
It seems like we are not calling `wg.Wait()` anywhere. Are we maybe missing that or could we just remove the waitgroup completely?

Reply:
Looks like it was left in after the simplification. I'll remove it